controllers/taloscontrolplane_controller.go

```diff
@@ -736,77 +790,30 @@ func (r *TalosControlPlaneReconciler) addClusterOwnerToObj(ctx context.Context,
 	return objPatchHelper.Patch(ctx, obj)
 }
 
-func (r *TalosControlPlaneReconciler) ensureNodesBooted(ctx context.Context, cluster *capiv1.Cluster, machines []capiv1.Machine) error {
-	clientset, err := r.kubeconfigForCluster(ctx, util.ObjectKey(cluster))
-	if err != nil {
-		return err
-	}
-
-	client, err := r.talosconfigForMachines(ctx, clientset, machines...)
-	if err != nil {
-		return err
-	}
-
-	ctx, cancel := context.WithTimeout(ctx, time.Second*5)
-
-	nodesBootStarted := map[string]struct{}{}
-	nodesBootStopped := map[string]struct{}{}
-
-	err = client.EventsWatch(ctx, func(ch <-chan talosclient.Event) {
-		defer cancel()
-
-		for event := range ch {
-			if msg, ok := event.Payload.(*machineapi.SequenceEvent); ok {
-				if msg.GetSequence() == "boot" { // can't use runtime constants as they're in `internal/`
-					switch msg.GetAction() { //nolint:exhaustive
-					case machineapi.SequenceEvent_START:
-						nodesBootStarted[event.Node] = struct{}{}
-					case machineapi.SequenceEvent_STOP:
-						nodesBootStopped[event.Node] = struct{}{}
-					}
-				}
-			}
-		}
-	}, talosclient.WithTailEvents(-1))
-
-	if err != nil {
-		unwrappedErr := err
-
-		for {
-			if s, ok := status.FromError(unwrappedErr); ok && s.Code() == codes.DeadlineExceeded {
-				// ignore deadline exceeded as we've just exhausted events list
-				err = nil
-
-				break
-			}
-
-			unwrappedErr = errors.Unwrap(unwrappedErr)
-			if unwrappedErr == nil {
-				break
-			}
-		}
-	}
-
-	if err != nil && !errors.Is(err, context.Canceled) {
-		return err
-	}
-
-	nodesNotFinishedBooting := []string{}
-
-	// check for nodes which have Boot/Start event, but no Boot/Stop even
-	// if the node is up long enough, Boot/Start even might get out of the window,
-	// so we can't check such nodes reliably
-	for node := range nodesBootStarted {
-		if _, ok := nodesBootStopped[node]; !ok {
-			nodesNotFinishedBooting = append(nodesNotFinishedBooting, node)
-		}
-	}
-
-	sort.Strings(nodesNotFinishedBooting)
+func patchTalosControlPlane(ctx context.Context, patchHelper *patch.Helper, tcp *controlplanev1.TalosControlPlane, opts ...patch.Option) error {
+	// Always update the readyCondition by summarizing the state of other conditions.
+	conditions.SetSummary(tcp,
+		conditions.WithConditions(
+			clusterv1.ReadyCondition,
+			controlplanev1.ResizedCondition,
+			controlplanev1.MachinesReadyCondition,
+			controlplanev1.AvailableCondition,
```

Comment:
should we have all conditions here?

Reply:
I've used this as a reference. Though I forgot to add couple conditions.